File: enode/src/main/java/org/enodeframework/messaging/IMessageDispatcher.java
package org.enodeframework.messaging;

import java.util.List;
import java.util.concurrent.CompletableFuture;

public interface IMessageDispatcher {
    CompletableFuture<Void> dispatchMessageAsync(IMessage message);

    CompletableFuture<Void> dispatchMessagesAsync(List<? extends IMessage> messages);
}

File: enode/src/main/java/org/enodeframework/messaging/IMessageDispatcher.kt
package org.enodeframework.messaging

import java.util.concurrent.CompletableFuture

interface IMessageDispatcher {
    fun dispatchMessageAsync(message: IMessage): CompletableFuture<Boolean>
    fun dispatchMessagesAsync(messages: List<IMessage>): CompletableFuture<Boolean>
}